### PR TITLE
Integration company list with service

### PR DESCRIPTION
**Pull request by vdorceus-excilys**
Integration of the company service developed by Flora with the visual component for listing companies (with filter and search support)

### CODE REVIEW

File: src/app/list-company/list-company.component.ts [hunk 1]
```diff
@@ -28,16 +29,27 @@ export class ListCompanyComponent implements OnInit {
...
-    this.dataSource = new MatTableDataSource(companyList);
+
+  constructor(private companyService: CompanyService,private changeDetector: ChangeDetectorRef) {     
+    
   }
 
   ngOnInit() {
+    this.companyService.getCompanies().subscribe(
+      data => {
+        console.log(data);
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.ts]: Remove log

File: src/app/list-company/list-company.component.ts [hunk 2]
```diff
@@ -58,3 +70,13 @@ function createCompany(id: number): CompanyDTO{
   console.log(myCompany);
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.ts]: Remove log

File: src/app/list-company/list-company.component.ts [hunk 3]
```diff
@@ -58,3 +70,13 @@ function createCompany(id: number): CompanyDTO{
...
   return myCompany;
 }
+
+/** Builds and returns a new User. */
+function createCompany(id: number): CompanyDTO{
```
Comment (RoudautTanguy) [inline src/app/list-company/list-company.component.ts]: Never used

Comment (vdorceus-excilys) [reply src/app/list-company/list-company.component.ts]: fixed

File: src/app/list-company/list-company.component.ts (hunk 2 above)
Comment (vdorceus-excilys) [reply src/app/list-company/list-company.component.ts]: removed

File: src/app/list-company/list-company.component.ts (hunk 1 above)
Comment (vdorceus-excilys) [reply src/app/list-company/list-company.component.ts]: removed

**Pull request merged**